### PR TITLE
feat(functions): Add regionOrCustomDomain option to useEmulator.

### LINKED ISSUE #765: bug: Fix on Web and add argument for Region Support to FirebaseFunctions.useEmulator for Web Version
### Plugin(s)

- [ ] Analytics
- [ ] App
- [ ] App Check
- [ ] Authentication
- [ ] Crashlytics
- [ ] Cloud Firestore
- [X] Cloud Functions
- [ ] Cloud Messaging
- [ ] Cloud Storage
- [ ] Performance
- [ ] Remote Config

### Version

6.2.0

### Platform(s)

- [ ] Android
- [ ] iOS
- [X] Web

### Current behavior

**Package**: `@capacitor-firebase/functions`
**Method**: `FirebaseFunctions.useEmulator`

When using the `useEmulator` method for the web version, an error occurs due to the absence of certain required arguments in `getFunctions()`. Currently, the `getFunctions()` method does not accept the `app?: FirebaseApp` argument, which causes issues.

Additionally, the emulator defaults to the `us-central1` region, and there is no way to specify a different region or domain. This leads to problems when Cloud Functions are deployed in regions other than `us-central1`.

### Expected behavior

- The `getFunctions()` method should accept an optional `app` argument to properly configure the emulator.
- There should be an option to specify the region or domain in the `useEmulator` method to ensure compatibility with Cloud Functions deployed in other regions.

### Reproduction

1. Use `FirebaseFunctions.useEmulator` in a web environment.
2. Dev a Cloud Function in a region other than `us-central1`.
3. Attempt to access the Cloud Function through the emulator.
4. Observe error indicating that the cannot find the function and see public link not local emulator link for cloud function.

### Other information

This issue is similar to the one described in [Firebase JS SDK Issue #2510](https://github.com/firebase/firebase-js-sdk/issues/2510).

Thank you for considering these suggestions to improve the compatibility of the emulator with Cloud Functions located in different regions.

### Capacitor doctor

💊   Capacitor Doctor  💊 

Latest Dependencies:

  @capacitor/cli: 6.2.0
  @capacitor/core: 6.2.0
  @capacitor/android: 6.2.0
  @capacitor/ios: 6.2.0

Installed Dependencies:

  @capacitor/cli: 6.0.0
  @capacitor/ios: 6.0.0
  @capacitor/core: 6.0.0
  @capacitor/android: 6.0.0

### Before submitting

- [X] I have read and followed the [bug report guidelines](https://capawesome.io/contributing/bug-reports/).
- [X] I have attached links to possibly related issues and discussions.
- [X] I understand that incomplete issues (e.g. without reproduction) are closed.

### PR DESCRIPTION
**Pull request by dimer47**
* fix useEmulator for specific region

Introduced a new property `regionOrCustomDomain` to the `UseEmulatorOptions` interface to enable specifying either a region or a custom domain for callable Firebase functions. This change facilitates greater flexibility in configuring the functions emulator based on deployment preferences. Updated related documentation to reflect these changes for clarity.

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] The changes have been tested successfully.
- [x] A changeset has been created (`npm run changeset`).
- [x] I have read and followed the issue : https://github.com/capawesome-team/capacitor-firebase/issues/765

Close #765

### CODE REVIEW

File: packages/functions/src/web.ts [hunk 1]
```diff
@@ -48,7 +50,7 @@ export class FirebaseFunctionsWeb
   }
 
   public async useEmulator(options: UseEmulatorOptions): Promise<void> {
-    const functions = getFunctions();
+    const functions = getFunctions(getApp(), options.regionOrCustomDomain);
```
Comment (robingenz) [inline packages/functions/src/web.ts:53]:
```suggestion
    const functions = getFunctions(undefined, options.regionOrCustomDomain);
```

Is the `getApp()` call necessary? If we set undefined it should automatically take the default app (I haven't tested it, just a thought).

File: packages/functions/src/definitions.ts [hunk 2]
```diff
@@ -123,4 +123,10 @@ export interface UseEmulatorOptions {
    * @example 5001
    */
   port?: number;
+  /**
+   * one of:
+   *   a) The region the callable functions are located in (ex: us-central1)
+   *   b) A custom domain hosting the callable functions (ex: https://mydomain.com)
+   */
```
Comment (robingenz) [inline packages/functions/src/definitions.ts:130]:
```suggestion
  /**
   * The region the callable functions are located in or a custom domain hosting the callable functions.
   * 
   * @example 'us-central1'
   * @example 'https://mydomain.com'
   */
```

File: .changeset/cuddly-baboons-add.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@capacitor-firebase/functions': major
```
Comment (robingenz) [inline .changeset/cuddly-baboons-add.md:2]:
```suggestion
'@capacitor-firebase/functions': minor
```

File: .changeset/cuddly-baboons-add.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
...
+---
+
+fix(functions): Add regionOrCustomDomain option to useEmulator.
```
Comment (robingenz) [inline .changeset/cuddly-baboons-add.md:5]:
```suggestion
fix(functions): add `regionOrCustomDomain` option to `UseEmulatorOptions`
```

**Review by robingenz — CHANGES_REQUESTED** [review]
Just a few comments.

File: packages/functions/src/web.ts (hunk 1 above)
Comment (dimer47) [reply packages/functions/src/web.ts]: Yes, absolutely true, @robingenz. I hadn’t noticed that this parameter was optional. Indeed, without defining it, it automatically takes the default app.

**Comment by dimer47** [comment]
Hi @robingenz,

Thank you for review! 
I've made the modification and will push it again shortly for those concerned.

File: .changeset/cuddly-baboons-add.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+'@capacitor-firebase/functions': minor
+---
+
+fix(functions): add `regionOrCustomDomain` option to `UseEmulatorOptions`
```
Comment (robingenz) [inline .changeset/cuddly-baboons-add.md:5]:
```suggestion
feat(functions): add `regionOrCustomDomain` option to `UseEmulatorOptions`
```

**Review by robingenz — APPROVED** [review]
Thank you!

**Review by robingenz — CHANGES_REQUESTED** [review]
Please run `npm run fmt` to format the code.

**Comment by robingenz** [comment]
@dimer47 Let me know if you have any questions!

**Comment by dimer47** [comment]
@robingenz sorry for the delay in getting back, I formatted the code and amended my commit with the correction.

File: .changeset/cuddly-baboons-add.md (hunk 5 above)
Comment (robingenz) [inline .changeset/cuddly-baboons-add.md:5]:
```suggestion
feat(functions): add `regionOrCustomDomain` option to `UseEmulatorOptions`
```

**Review by robingenz — APPROVED** [review]
Thank you!

**Pull request merged**